components/sensor/adxl345/adxl345.go

```diff
@@ -0,0 +1,196 @@
+			cfg config.Component,
+			logger golog.Logger,
+		) (interface{}, error) {
+			return NewAdxl345(ctx, deps, cfg, logger)
+		},
+	})
+
+	config.RegisterComponentAttributeMapConverter(sensor.SubtypeName, modelName,
+		func(attributes config.AttributeMap) (interface{}, error) {
+			var attr AttrConfig
+			return config.TransformAttributeMapToStruct(&attr, attributes)
+		},
+		&AttrConfig{})
+}
+
+type adxl345 struct {
+	bus        board.I2C
+	i2cAddress byte
+	mu         sync.Mutex
+	logger     golog.Logger
+
+	generic.Unimplemented // Implements DoCommand with an ErrUnimplemented response
+}
+
+// NewAdxl345 is a constructor to create a new object representing an ADXL345 accelerometer.
+func NewAdxl345(
+	ctx context.Context,
+	deps registry.Dependencies,
+	rawConfig config.Component,
+	logger golog.Logger,
+) (sensor.Sensor, error) {
+	cfg := rawConfig.ConvertedAttributes.(*AttrConfig)
+	b, err := board.FromDependencies(deps, cfg.BoardName)
+	if err != nil {
+		return nil, err
+	}
+	localB, ok := b.(board.LocalBoard)
+	if !ok {
+		return nil, errors.Errorf("board %s is not local", cfg.BoardName)
+	}
+	bus, ok := localB.I2CByName(cfg.BusID)
+	if !ok {
+		return nil, errors.Errorf("can't find I2C bus '%s' for ADXL345 sensor", cfg.BusID)
+	}
+
+	var address byte
+	if cfg.UseAlternateI2CAddress {
+		address = 0x1D
+	} else {
+		address = 0x53
+	}
+
+	sensor := &adxl345{
+		bus:        bus,
+		i2cAddress: address,
+		logger:     logger,
+	}
+
+	// To check that we're able to talk to the chip, we should be able to read register 0 and get
+	// back the device ID (0xE5).
+	deviceID, err := sensor.readByte(ctx, 0)
+	if err != nil {
+		return nil, errors.Errorf("can't read from I2C address %d on bus %s of board %s: '%s'",
+			address, cfg.BusID, cfg.BoardName, err.Error())
+	}
+	if deviceID != 0xE5 {
+		return nil, errors.Errorf("unexpected I2C device instead of ADXL345 at address %d: deviceID '%d'",
+			address, deviceID)
+	}
+
+	// The chip starts out in standby mode. Set it to measurement mode so we can get data from it.
+	// To do this, we set the Power Control register (0x2D) to turn on the 8's bit.
+	err = sensor.writeByte(ctx, 0x2D, 0x08)
+	if err != nil {
+		return nil, errors.Errorf("unable to put ADXL345 into measurement mode: '%s'", err.Error())
+	}
+
+	return sensor, nil
+}
+
+func (adxl *adxl345) readByte(ctx context.Context, register byte) (byte, error) {
+	result, err := adxl.readBlock(ctx, register, 1)
+	if err != nil {
+		return 0, err
+	}
+	return result[0], err
+}
+
+func (adxl *adxl345) readBlock(ctx context.Context, register byte, length uint8) ([]byte, error) {
+	handle, err := adxl.bus.OpenHandle(adxl.i2cAddress)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		err := handle.Close()
+		if err != nil {
+			adxl.logger.Error(err)
+		}
+	}()
+
+	results, err := handle.ReadBlockData(ctx, register, length)
+	return results, err
+}
+
+func (adxl *adxl345) writeByte(ctx context.Context, register, value byte) error {
```

Comment:
just for consistency would nice to have a flow similar to Read

Reply:
Tell me more. If you're saying there should be a `writeBlock` function analogous to `readBlock`, I don't like it because it would be an unused private function, which is dead code. If you're saying `writeByte` should look similar to `readByte`, the tweak around that is that `readByte is just a wrapper around `readBlock`, and this function looks similar to `readBlock` already. or maybe you're saying something else?